core/src/HIP/Kokkos_HIP_Parallel_Range.hpp

```diff
@@ -614,7 +602,10 @@ class ParallelScanHIPBase {
   }
 
   // Determine block size constrained by shared memory:
-  virtual inline unsigned local_block_size(const FunctorType& f) = 0;
+  virtual inline unsigned local_block_size(const FunctorType&) {
+    // Making this function pure virtual, trips the compiler.
+    return 0;
```

Comment:
hmm... as we can not make it pure, can we at least throw an `exception` or an `assert` ... just for sanity ... should never be reached anyways, but if it does we would see it

Reply:
I  would just get rid of this function and pass the block size directly to the base class.

Reply:
we already throw if the block size is zero